File: files/CH-151.R
library(tidyverse)
library(readxl)

path = "files/CH-151 Solid Worl.xlsx"
input = read_excel(path, range = "B2:F10")
test  = read_excel(path, range = "H2:J11") %>%
  mutate(across(c("y", "z")), round(., 5))

result = input %>%
  as.matrix() %>% 
  t() %>%
  as.numeric() %>%
  keep(!is.na(.)) %>%
  matrix(ncol = 3, byrow = TRUE) %>%
  as.data.frame() %>%
  setNames(c("x", "y", "z")) %>%
  mutate(across(c("y", "z")), round(., 5))

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/100-199/151/CH-151.py
import pandas as pd
import numpy as np

path = "CH-151 Solid Worl.xlsx"

input = pd.read_excel(path, usecols="B:F", skiprows=1, nrows=8).to_numpy().flatten()
test = pd.read_excel(path, usecols="H:J", skiprows=1, nrows=9)

result = [x for x in input if isinstance(x, (int, float)) and not pd.isna(x)]
matrix = np.array(result).reshape(-1, 3)
result_df = pd.DataFrame(matrix, columns=["x", "y", "z"])

print(all(result_df == test)) # True
